src/internet_identity_interface/src/lib.rs

```diff
@@ -173,3 +174,90 @@ pub struct HttpResponse {
 pub struct InternetIdentityInit {
     pub assigned_user_number_range: (UserNumber, UserNumber),
 }
+
+// Archive specific types
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub enum OperationType {
+    #[serde(rename = "register_anchor")]
+    RegisterAnchor { device: DeviceDataWithoutAlias },
+    #[serde(rename = "add_device")]
+    AddDevice { device: DeviceDataWithoutAlias },
+    #[serde(rename = "update_device")]
+    UpdateDevice {
+        device: PublicKey,
+        new_values: DeviceDataUpdate,
+    },
+    #[serde(rename = "remove_device")]
+    RemoveDevice { device: PublicKey },
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub struct Entry {
+    // store anchor in LogEntry, such that anchor operations can be attributed to an anchor without consulting the index.
+    pub anchor: Anchor,
+    pub operation: OperationType,
+    pub timestamp: Timestamp,
+    pub caller: Principal,
+    // global sequence number to detect lost messages (if any)
+    pub sequence_number: u64,
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub struct DeviceDataWithoutAlias {
+    pub pubkey: DeviceKey,
+    pub credential_id: Option<CredentialId>,
+    pub purpose: Purpose,
+    pub key_type: KeyType,
+    pub protection: DeviceProtection,
+}
+
+// If present, the attribute has been changed to the value given.
+// Does not include the pubkey because it cannot be changed.
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub struct DeviceDataUpdate {
+    pub alias: Option<Private>,
+    pub credential_id: Option<CredentialId>,
+    pub purpose: Option<Purpose>,
+    pub key_type: Option<KeyType>,
+    pub protection: Option<DeviceProtection>,
+}
+
+// Placeholder for information that has been hidden for privacy reasons.
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub enum Private {
+    HiddenForPrivacyReasons,
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize)]
+pub struct Entries {
```

Comment:
`Entries` itself (and `AnchorEntries`) is never serialized to stable memory, right? why do we care about keeping this extensible?